mealie/core/security/security.py

```diff
@@ -128,10 +129,35 @@ def user_from_ldap(db: AllRepositories, username: str, password: str) -> Private
     return user
 
 
-def authenticate_user(session, email: str, password: str) -> PrivateUser | bool:
+def _create_new_jwt_user(db: AllRepositories, claims: dict) -> PrivateUser:
+    settings = get_app_settings()
+    return db.users.create(
+        {
+            "full_name": claims[settings.JWT_AUTH_NAME_CLAIM],
+            "username": claims[settings.JWT_AUTH_USERNAME_CLAIM],
+            "email": claims[settings.JWT_AUTH_EMAIL_CLAIM],
+            "password": hash_password(secrets.token_urlsafe(13)),  # 13 char long random password
+            "group": settings.DEFAULT_GROUP,
+            "admin": True,
+        }
+    )
+
+
+def authenticate_user(session, email: str, password: str, jwt_assertion: str | None = None) -> PrivateUser | bool:
```

Comment:
This function is now optionally receiving a `jwt_assertion` which would be the encoded JWT if present.